### ISSUE TITLE
Analog stick extreme latency with Evdev input method

### ISSUE THREAD

**Issue by TriVoxel** [RPCS3/rpcs3#11609]
Linux AMD64 using native Appimage build

Analog sticks have extremely bad latency with Evdev input. It's okay with a small amount of movement, however creating a lot of movement will cause controlls to be very unresponsive for a long time.

Annoying video:
https://youtu.be/UkDtEj8G4OY

Video demonstrates this well, but it was much worse when I tried it yesterday. I haven't updated it since yesterday, so it seems somewhat unpredictable.

FYI: Filtering noise in controller configs makes it worse

**Comment by Megamouse** [comment]
Missing log file

**Comment by dogtopus** [comment]
Reproducible as of official release 39201e5042fd0d57f6b65fea1a26b397d970f49b and flatpak release `$official-2`.

It also seems that the native driver lags as well, but less so compare to evdev (native lags about 500ms while evdev is pretty much unusable).

It also doesn't matter if the transport is USB or Bluetooth, the device is DS4 or DS5. They all lag the same way.

OS is Fedora Silverblue 35.

Logs from the official release: [RPCS3.log](https://github.com/RPCS3/rpcs3/files/8309949/RPCS3.log)

**Comment by Augusto7743** [comment]
Evdev is exactly udev ?
Ubuntu 20.04 using Retroarch with udev have high latency in keyboard input.

**Comment by Megamouse** [comment]
I also noticed these particular stick issues back in the day when I helped to implement this.
The issue is not that the sticks are slow or bugged. It's a simple performance issue in the pad thread. 

I might've fixed it in the mentioned pull request. Please give that one a shot.

**Comment by Megamouse** [comment]
To explain in some more detail, there's four possible ways to improve this:
1. First one is obvious, but not really helpful nor user friendly: Get faster hardware... xD
2. Use more threads to do more work concurrently. We already do this to some extend.
3. Increase the interval at which we query the events.
4. Try to cramp in more events in each interval.

So far, we did the following:
We first fetched one event for each of the buttons or axis. We then processed all these events, waited a millisecond so that your cpu is not unnecessarily occupied, and then do it all over again as long as the game is running.
This means that we processed each button or axis a thousand times per second at most.
In the pad settings dialog we do the same but only 100 times per second, because it doesn't really matter if it's 100% responsive in that dialog and GUI updates are expensive.

If your CPU can't handle these tasks fast enough, you will get some lag, especially if you have 2 players or more connected. But that's true for any given task in software. In our case, this occurs with the sticks, since they spam a lot of events while moving around. RPCS3 tries to work through all of the buffered events as fast as it can. So you kinda see it in slow motion.

There's already some secret sauce implemented in RPCS3 that can help you out, but I'd like you to test the pull request first, before we try that one.
